sdk/search/search-documents/test/public/typeDefinitions.ts

```diff
@@ -49,14 +42,13 @@ type ExpectKnownTokenFilterNames = `${KnownTokenFilterName}`;
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-ignore
 function fun() {
-  const a: IsIdentical<ExpectSearchFieldDataType, SearchFieldDataType> = "pass";
-  const b: IsIdentical<ExpectVectorSearchAlgorithmMetric, VectorSearchAlgorithmMetric> = "pass";
-  const c: IsIdentical<ExpectVectorSearchAlgorithmKind, VectorSearchAlgorithmKind> = "pass";
-  const d: IsIdentical<ExpectVectorFilterMode, VectorFilterMode> = "pass";
-  const e: IsIdentical<ExpectKnownCharFilterNames, `${KnownCharFilterNames}`> = "pass";
-  const f: IsIdentical<ExpectKnownAnalyzerNames, `${KnownAnalyzerNames}`> = "pass";
-  const g: IsIdentical<ExpectKnownTokenizerNames, `${KnownTokenizerNames}`> = "pass";
-  const h: IsIdentical<ExpectKnownTokenFilterNames, `${KnownTokenFilterNames}`> = "pass";
-
-  return [a, b, c, d, e, f, g, h];
+  const a: IsIdentical<ExpectVectorSearchAlgorithmMetric, VectorSearchAlgorithmMetric> = "pass";
+  const b: IsIdentical<ExpectVectorSearchAlgorithmKind, VectorSearchAlgorithmKind> = "pass";
```

Comment:
You should give us a demo of these type-based assertions! Not to put you on the spot, but personally I'd love to hear more about it

Reply:
I remember someone else was trying to tackle breaking change detection, so this stint in generating type assertions was short-lived. If it evolves into something more usable, I'll definitely share with the team!